### PR TITLE
feat: add monitor_from_point method, closes #187

### LINKED ISSUE #187: feat: add `monitor::from_point()`
Add a method called `monitor::from_point()`.  Apologies I am not 100% sure what this method is for, but it was discussed by @amrbashir in [this thread.](https://github.com/tauri-apps/tauri/issues/2258)

Possibly for allow calculations to position windows between workspaces / monitors.

### PR DESCRIPTION
**Pull request by CNLHC**
### What kind of change does this PR introduce?

- [x] Feature

### Does this PR introduce a breaking change?

- [x] No

### Checklist
- [x] When resolving issues, they are referenced in the PR's title (e.g `fix: remove a typo, closes #___, #___`)
- [ ] A change file is added if any packages will require a version bump due to this PR per [the instructions in the readme](https://github.com/tauri-apps/tauri/blob/dev/.changes/readme.md).
- [ ] I have added a convincing reason for adding this feature, if necessary

### CODE REVIEW

**Comment by CNLHC** [comment]
Some discussions are posted to the https://github.com/tauri-apps/tao/issues/187

File: src/window.rs [hunk 1]
```diff
@@ -1184,6 +1184,11 @@ impl Window {
     self.window.current_monitor()
   }
 
+  #[inline]
+  pub fn monitor_from_point(&self, x: f64, y: f64) -> Option<MonitorHandle> {
+    self.window.monitor_from_point(x, y)
+  }
+
```
Comment (amrbashir) [inline src/window.rs]: `monitor::from_point` main usage will be without a Window, so we also needs it to be exposed on `event_loop::EventLoopWindowTarget`

File: src/platform_impl/macos/monitor.rs [hunk 2]
```diff
@@ -158,6 +158,27 @@ pub fn primary_monitor() -> MonitorHandle {
...
+        && y >= bound.origin.y
+        && y <= bound.origin.y + bound.size.height
+      {
+        return Some(monitor);
+      }
+    }
+  }
+  return None;
+}
+
```
Comment (amrbashir) [inline src/platform_impl/macos/monitor.rs]: Windows and Linux have a dedicated API for these and I am not sure if macOS has one or not but if it doesn't, this is fine.

- Linux: https://gtk-rs.org/gtk3-rs/stable/latest/docs/gdk/struct.Display.html#method.monitor_at_point
- Widnows: https://learn.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-monitorfrompoint

**Review by amrbashir — CHANGES_REQUESTED** [review]

File: src/platform_impl/macos/monitor.rs (hunk 2 above)
Comment (CNLHC) [reply src/platform_impl/macos/monitor.rs]: Thanks for your suggestion. I will use these two method for windows and linux

File: src/window.rs (hunk 1 above)
Comment (CNLHC) [reply src/window.rs]: sure

corresponding methods have added into this PR

File: src/platform_impl/windows/monitor.rs [hunk 3]
```diff
@@ -123,6 +123,23 @@ pub fn current_monitor(hwnd: HWND) -> MonitorHandle {
   MonitorHandle::new(hmonitor)
 }
 
+pub fn from_points(x: f64, y: f64) -> Option<MonitorHandle> {
```
Comment (amrbashir) [inline src/platform_impl/windows/monitor.rs]:
```suggestion
pub fn from_point(x: f64, y: f64) -> Option<MonitorHandle> {
```

File: src/platform_impl/windows/monitor.rs [hunk 4]
```diff
@@ -132,6 +149,10 @@ impl Window {
     let monitor = primary_monitor();
     Some(RootMonitorHandle { inner: monitor })
   }
+
+  pub fn monitor_from_point(&self, x: f64, y: f64) -> Option<RootMonitorHandle> {
+    from_points(x, y).map(|inner| RootMonitorHandle { inner })
```
Comment (amrbashir) [inline src/platform_impl/windows/monitor.rs]:
```suggestion
    from_point(x, y).map(|inner| RootMonitorHandle { inner })
```

**Review by amrbashir — APPROVED** [review]
Thank you for your contribution. A couple of nit-picks and should be read for merge.

**Review by amrbashir — APPROVED** [review]

**Comment by CNLHC** [comment]
I have a question
how should we implement  such functions for ios/android?

**Comment by amrbashir** [comment]
Just add empty functions. It is not supported on Android/IOS, sorry I shoulve caught that.

File: src/event_loop.rs [hunk 5]
```diff
@@ -213,6 +213,15 @@ impl<T> EventLoopWindowTarget<T> {
     self.p.primary_monitor()
   }
 
+  /// Returns the monitor that contains the given point.
```
Comment (amrbashir) [inline src/event_loop.rs:216]:
```suggestion
  /// Returns the monitor that contains the given point.
  ///
  /// ## Platform-specific:
  ///
  /// - **Android / iOS:** Unsupported.
```

File: src/window.rs [hunk 6]
```diff
@@ -1184,6 +1184,11 @@ impl Window {
     self.window.current_monitor()
   }
 
+  #[inline]
```
Comment (amrbashir) [inline src/window.rs:1187]:
```suggestion
  #[inline]
  /// Returns the monitor that contains the given point.
  ///
  /// ## Platform-specific:
  ///
  /// - **Android / iOS:** Unsupported.
```

**Review by amrbashir — CHANGES_REQUESTED** [review]

**Comment by CNLHC** [comment]
> Just add empty functions. It is not supported on Android/IOS, sorry I shoulve caught that.

Ok, I will add an empty function to these two platforms.

**Comment by CNLHC** [comment]
I have added stub functions to the Android/IOS platform, would you please review this PR again @amrbashir?

File: src/platform_impl/linux/event_loop.rs [hunk 7]
```diff
@@ -57,6 +57,11 @@ pub struct EventLoopWindowTarget<T> {
 }
 
 impl<T> EventLoopWindowTarget<T> {
+  #[inline]
+  pub fn monitor_from_point(&self, x: f64, y: f64) -> Option<MonitorHandle> {
+    let display = &self.display;
+    monitor::from_point(display, x, y)
```
Comment (amrbashir) [inline src/platform_impl/linux/event_loop.rs]:
```suggestion
    monitor::from_point(&self.display, x, y)
```

File: src/platform_impl/macos/monitor.rs [hunk 8]
```diff
@@ -158,6 +158,27 @@ pub fn primary_monitor() -> MonitorHandle {
...
 }
 
+// `from_point` get a monitor handle which contains the given point.
+pub fn from_point(x: f64, y: f64) -> Option<MonitorHandle> {
+  unsafe {
+    for monitor in available_monitors() {
+      let bound = CGDisplayBounds(monitor.0);
+      // dumb way. a better way is to use [CGRectContainsPoint](https://developer.apple.com/documentation/coregraphics/1456316-cgrectcontainspoint)
+      // which is not supported by core-graphics binding yet.
+      // Once the [PR](https://github.com/servo/core-foundation-rs/pull/526) is accepted
```
Comment (amrbashir) [inline src/platform_impl/macos/monitor.rs]: you can inline the implementation of that PR here instead. cocoa-rs will not release anytime soon.

**Review by amrbashir — CHANGES_REQUESTED** [review]

File: src/platform_impl/macos/monitor.rs (hunk 8 above)
Comment (CNLHC) [reply src/platform_impl/macos/monitor.rs]: done

**Review by amrbashir — APPROVED** [review]

**Review by wusyong — APPROVED** [review]

**Comment by CNLHC** [comment]
Thanks for your review @wusyong

**Pull request merged**